tests/components/test-class-component.php

```diff
@@ -0,0 +1,624 @@
+		);
+	}
+
+	/**
+	 * Data provider for test_get_theme
+	 *
+	 * @return array[] Array of test arguments.
+	 */
+	public function provide_get_theme_data() {
+		return [
+			[
+				[],
+				'default',
+				'Default theme not correct.',
+			],
+			[
+				[ 'theme' => 'foo' ],
+				'foo',
+				'Invalid theme should fall back to defaults.',
+			],
+			[
+				[ 'theme' => 'primary' ],
+				'primary',
+				'Primary theme not correct.',
+			],
+			[
+				[ 'theme' => 'secondary' ],
+				'secondary',
+				'Secondary theme not correct.',
+			],
+		];
+	}
+
+	/**
+	 * Test that context is applied correctly.
+	 *
+	 * @group context
+	 */
+	public function test_get_context() {
+		get_context_store()->set(
+			[ 'test/foo' => 'bar' ]
+		);
+
+		// 'test/use-context' is a registered type.
+		$component = new Component( 'test/use-context' );
+
+		// Clean up.
+		get_context_store()->reset();
+
+		$this->assertSame(
+			[
+				'test/foo' => 'bar',
+				'test/baz' => null,
+			],
+			$component->get_context(),
+			'Calculated context values not set.'
+		);
+	}
+
+	/**
+	 * Test that context is applied correctly.
+	 *
+	 * @dataProvider provide_context_config_data
+	 * @group context
+	 *
+	 * @param array $config   Test config values.
+	 * @param array $expected Expected value.
+	 */
+	public function test_context_values_set( $config, $expected ) {
+		get_context_store()->set(
+			[ 'test/foo' => 'bar' ]
+		);
+
+		// 'test/use-context' is a registered type.
+		$component = new Component(
+			'test/use-context',
+			[ 'config' => $config ]
+		);
+
+		// Clean up.
+		get_context_store()->reset();
+
+		$this->assertSame(
+			$expected,
+			$component->get_config()
+		);
+	}
+
+	/**
+	 * Data provider for test_context_values_set
+	 *
+	 * @return array[]
+	 */
+	public function provide_context_config_data() {
+		return [
+			[
+				[],
+				[
+					'foo' => 'bar',
+					'baz' => 'default',
+				],
+			],
+			[
+				[
+					'foo' => 'overridden',
+				],
+				[
+					'foo' => 'overridden',
+					'baz' => 'default',
+				],
+			],
+		];
+	}
+
+	/**
+	 * Test config callbacks.
+	 */
+	public function test_config_callback() {
+		$name = 'example/component';
+
+		/*
+		 * Register component with callback.
+		 *
+		 * We're using a callback that uses the config in order
+		 * to ensure that the callback is passed an array.
+		 */
+		get_registry()->register_component(
+			$name,
+			[
+				'config_callback' => function ( array $config ) {
+					return array_merge(
+						[ 'foo' => 'bar' ],
+						$config
+					);
+				},
+			]
+		);
+
+		$component = new Component(
+			$name,
+			[
+				'config' => [
+					'baz' => 'qux',
+				],
+			]
+		);
+
+		// Clean up.
+		get_registry()->unregister_component( $name );
+
+		$this->assertEquals(
+			[
+				'foo' => 'bar',
+				'baz' => 'qux',
+			],
+			$component->get_config()
+		);
+	}
+
+	/**
+	 * Test children set from config.
+	 */
+	public function test_children_from_config() {
+		// Test both array and string notation.
```

Comment:
Unsure if we really want to support both notation types or if we want to enforce the array syntax 🤷‍♂️

Reply:
Can you elaborate on what dot notion in this situation is?

Reply:
I was thinking it might be nice to support passing a list of component names as children, rather than making each be an array.